### PR TITLE
TCCP: Tooltip accessibility improvements

### PR DESCRIPTION
**Pull request by contolini**
Adds some code to close tooltips when `esc` key is pressed per [digital.gov guidance](https://designsystem.digital.gov/components/tooltip/accessibility-tests/). Also adds an `aria-label` to card results that summarizes all the card details to assistive technologies.

## How to test this PR

1. Everything should look the same.
1. Hover over a tooltip and click `esc` to close it.
1. `yarn cypress run --spec test/cypress/integration/consumer-tools/credit-cards/explore-cards.cy.js`

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Future todos are captured in comments and/or tickets
- [x] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Check the current browser support cutoff list](https://cfpb.github.io/consumerfinance.gov/browser-support#current-browser-support-metrics) for browsers that are advisable
to prioritize for testing.

#### Accessibility

- [x] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [x] Screen reader friendly
- [x] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Does not introduce new lint warnings
- [x] Flexible from small to large screens

### CODE REVIEW

File: cfgov/tccp/jinja2/tccp/includes/card_list.html [hunk 1]
```diff
@@ -51,9 +51,23 @@
 <div class="o-card-group">
     <div class="o-card-group__cards">
         {%- for card in cards %}
+        {%- set card_purchase_apr = apr_range(
+            card.purchase_apr_for_tier_min,
+            card.purchase_apr_for_tier_max,
+            asterisk=true
```
Comment (niqjohnson) [inline cfgov/tccp/jinja2/tccp/includes/card_list.html:57]: How is the asterisk read by a screen reader when this is used in the `aria-label`? If it sounds weird, I was going to suggest using an `asterisk=false` version in the `aria-label`. But if it's also read awkwardly in the card content, maybe we can update the `apr_range` macro to make the asterisk `aria-hidden` everywhere so it's never read.

File: cfgov/tccp/jinja2/tccp/includes/card_list.html [hunk 2]
```diff
@@ -51,9 +51,23 @@
...
+            default=('See details' if card.annual_fee_estimated is none and card.periodic_fee_type else '$0')
+        ) -%}
         {%- set show_more = loop.index > 11 and ordering_by != 'product_name' -%}
         <article class="m-card m-card--tabular{% if show_more %} u-show-more{% endif %}">
-            <a href="{{ card.url }}" data-js-hook="behavior_ignore-link-targets" data-ignore-link-targets="[data-tooltip]">
+            <a
+                href="{{ card.url }}"
+                data-js-hook="behavior_ignore-link-targets"
+                data-ignore-link-targets="[data-tooltip]"
+                aria-label="{{ card.product_name }} from {{ card.institution_name }} with {{ card_purchase_apr }} purchase APR, {{ card_account_fee }} account fee, {{ card_rewards(card) }} card rewards{%- if card.requirements_for_opening or card.issued_by_credit_union -%}, eligibility requirements{%- endif -%}"
```
Comment (niqjohnson) [inline cfgov/tccp/jinja2/tccp/includes/card_list.html:69]: Should we add the data published date in the description so it reads something like "9.99% purchase APR as of December 31, 2023"? That might be overkill and make for a too long summary, though.

**Review by niqjohnson — APPROVED** [review]
Working great for me locally. Just had a few minor questions and possible suggestions, but nothing to hold up merging.

File: cfgov/tccp/jinja2/tccp/includes/card_list.html (hunk 1 above)
Comment (contolini) [reply cfgov/tccp/jinja2/tccp/includes/card_list.html:57]: Great point, the asterisk does not read well either on the results page or the card detail pages. I'm thinking we should either migrate to a [footnote format](https://www.jamesrmeyer.com/otherstuff/easy-footnotes-for-web-pages) (e.g. `[1]`) that links to the explanatory text *or* we can try replacing the asterisks with the full explanatory text for screenreaders only. I'll experiment and open another PR.

File: cfgov/tccp/jinja2/tccp/includes/card_list.html (hunk 2 above)
Comment (contolini) [reply cfgov/tccp/jinja2/tccp/includes/card_list.html:69]: I was originally including the date in the aria label but it felt very repetitive and obnoxious to repeat "as of December 31, 2023" over and over again as the user tabs through the cards. I figure the card detail page has all the juicy details so it's okay to keep the card summaries concise.

**Pull request merged**